x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/UnboundedGeoTileGridTiler.java

```diff
@@ -28,4 +30,20 @@ protected boolean validTile(int x, int y, int z) {
     protected long getMaxCells() {
         return maxTiles;
     }
+
+    @Override
+    protected int setValuesForFullyContainedTile(int xTile, int yTile, int zTile, GeoShapeCellValues values, int valuesIndex) {
+        final int splits = 1 << precision - zTile;
+        final int minX =xTile * splits;
+        final int maxX = xTile * splits + splits - 1;
+        final int minY = yTile * splits;
+        final int maxY = yTile * splits + splits - 1;
```

Comment:
Could you add some explanation in a comment on what's going on here?

Reply:
Done